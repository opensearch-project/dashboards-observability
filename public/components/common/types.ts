/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export interface FieldCapAttributes {
  type: string;
  searchable: boolean;
  aggregatable: boolean;
}

export interface FieldCapResponse {
  indices: string[];
  fields: Record<string, Record<string, FieldCapAttributes>>;
}
