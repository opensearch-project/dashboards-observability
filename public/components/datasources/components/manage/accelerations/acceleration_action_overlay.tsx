/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from 'react';
import { EuiOverlayMask, EuiConfirmModal, EuiCompressedFormRow, EuiCompressedFieldText } from '@elastic/eui';
import { CachedAcceleration } from '../../../../../../common/types/data_connections';
import {
  ACC_DELETE_MSG,
  ACC_VACUUM_MSG,
  ACC_SYNC_MSG,
  AccelerationActionType,
  getAccelerationName,
  getAccelerationFullPath,
} from './utils/acceleration_utils';

export interface AccelerationActionOverlayProps {
  isVisible: boolean;
  actionType: AccelerationActionType;
  acceleration: CachedAcceleration | null;
  dataSourceName: string;
  onCancel: () => void;
  onConfirm: () => void;
}

export const AccelerationActionOverlay: React.FC<AccelerationActionOverlayProps> = ({
  isVisible,
  actionType,
  acceleration,
  dataSourceName,
  onCancel,
  onConfirm,
}) => {
  const [confirmationInput, setConfirmationInput] = useState('');

  if (!isVisible || !acceleration) {
    return null;
  }

  const displayIndexName = getAccelerationName(acceleration);
  const displayFullPath = getAccelerationFullPath(acceleration, dataSourceName);

  let title = '';
  let description = '';
  let confirmButtonText = 'Confirm';
  let confirmEnabled = true;

  switch (actionType) {
    case 'vacuum':
      title = `Vacuum acceleration ${displayIndexName} on ${displayFullPath} ?`;
      description = ACC_VACUUM_MSG;
      confirmButtonText = 'Vacuum';
      confirmEnabled = confirmationInput === displayIndexName;
      break;
    case 'delete':
      title = `Delete acceleration ${displayIndexName} on ${displayFullPath} ?`;
      description = ACC_DELETE_MSG(displayIndexName);
      confirmButtonText = 'Delete';
      break;
    case 'sync':
      title = 'Manual sync data?';
      description = ACC_SYNC_MSG;
      confirmButtonText = 'Sync';
      break;
  }

  return (
    <EuiOverlayMask>
      <EuiConfirmModal
        title={title}
        onCancel={onCancel}
        onConfirm={() => onConfirm()}
        cancelButtonText="Cancel"
        confirmButtonText={confirmButtonText}
        buttonColor="danger"
        defaultFocusedButton="confirm"
        confirmButtonDisabled={!confirmEnabled}
      >
        <p>{description}</p>
        {actionType === 'vacuum' && (
          <EuiCompressedFormRow label={`To confirm, type ${displayIndexName}`}>
            <EuiCompressedFieldText
              name="confirmationInput"
              value={confirmationInput}
              onChange={(e) => setConfirmationInput(e.target.value)}
            />
          </EuiCompressedFormRow>
        )}
      </EuiConfirmModal>
    </EuiOverlayMask>
  );
};
