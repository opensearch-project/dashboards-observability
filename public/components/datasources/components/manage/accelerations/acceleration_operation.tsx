/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { useEffect, useState } from 'react';
import { CachedAcceleration } from '../../../../../../common/types/data_connections';
import { useToast } from '../../../../common/toast';
import { useDirectQuery } from '../../../../../framework/datasources/direct_query_hook';
import { DirectQueryLoadingStatus } from '../../../../../../common/types/explorer';
import {
  AccelerationActionType,
  generateAccelerationOperationQuery,
  getAccelerationName,
} from './utils/acceleration_utils';

export const useAccelerationOperation = (dataSource: string) => {
  const { startLoading, stopLoading, loadStatus } = useDirectQuery();
  const { setToast } = useToast();
  const [isOperating, setIsOperating] = useState(false);
  const [operationSuccess, setOperationSuccess] = useState(false);
  const [accelerationToOperate, setAccelerationToOperate] = useState<CachedAcceleration | null>(
    null
  );
  const [operationType, setOperationType] = useState<AccelerationActionType | null>(null);
  const [currentStatus, setCurrentStatus] = useState<DirectQueryLoadingStatus | null>(null);

  useEffect(() => {
    if (!accelerationToOperate || !operationType || loadStatus === currentStatus) return;

    const displayAccelerationName = getAccelerationName(accelerationToOperate);

    let operationInProgressMessage = '';
    let operationSuccessMessage = '';
    let operationFailureMessage = '';

    switch (operationType) {
      case 'delete':
        operationInProgressMessage = `Deleting acceleration: ${displayAccelerationName}`;
        operationSuccessMessage = `Successfully deleted acceleration: ${displayAccelerationName}`;
        operationFailureMessage = `Failed to delete acceleration: ${displayAccelerationName}`;
        break;
      case 'vacuum':
        operationInProgressMessage = `Vacuuming acceleration: ${displayAccelerationName}`;
        operationSuccessMessage = `Successfully vacuumed acceleration: ${displayAccelerationName}`;
        operationFailureMessage = `Failed to vacuum acceleration: ${displayAccelerationName}`;
        break;
      case 'sync':
        operationInProgressMessage = `Syncing acceleration: ${displayAccelerationName}`;
        break;
    }

    if (loadStatus === DirectQueryLoadingStatus.SCHEDULED && operationType !== 'sync') {
      setIsOperating(true);
      setToast(operationInProgressMessage, 'success');
    } else if (loadStatus === DirectQueryLoadingStatus.SUCCESS && operationType !== 'sync') {
      setIsOperating(false);
      setAccelerationToOperate(null);
      setOperationSuccess(true);
      setToast(operationSuccessMessage, 'success');
    } else if (loadStatus === DirectQueryLoadingStatus.FAILED && operationType !== 'sync') {
      setIsOperating(false);
      setOperationSuccess(false);
      setToast(operationFailureMessage, 'danger');
    } else if (operationType === 'sync' && loadStatus === DirectQueryLoadingStatus.SCHEDULED) {
      setToast(operationInProgressMessage, 'success');
      stopLoading();
    }

    setCurrentStatus(loadStatus);
  }, [loadStatus, setToast, accelerationToOperate, operationType, currentStatus]);

  useEffect(() => {
    return () => {
      stopLoading();
    };
  }, []);

  const performOperation = (
    acceleration: CachedAcceleration,
    operation: AccelerationActionType
  ) => {
    setOperationSuccess(false);
    setOperationType(operation);
    const operationQuery = generateAccelerationOperationQuery(acceleration, dataSource, operation);

    const requestPayload = {
      lang: 'sql',
      query: operationQuery,
      datasource: dataSource,
    };

    setIsOperating(true);
    setAccelerationToOperate(acceleration);
    startLoading(requestPayload);
  };

  return { performOperation, isOperating, operationSuccess };
};
