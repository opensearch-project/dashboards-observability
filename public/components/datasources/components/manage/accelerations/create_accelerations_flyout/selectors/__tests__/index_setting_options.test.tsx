/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/dom';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import { CreateAccelerationForm } from '../../../../../../../../../common/types/data_connections';
import { createAccelerationEmptyDataMock } from '../../../../../../../../../test/accelerations';
import { IndexSettingOptions } from '../index_setting_options';

describe('Index settings acceleration components', () => {
  configure({ adapter: new Adapter() });

  it('renders acceleration index settings with default options', async () => {
    const accelerationFormData = createAccelerationEmptyDataMock;
    const setAccelerationFormData = jest.fn();
    const wrapper = mount(
      <IndexSettingOptions
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });

  it('renders acceleration index settings with different options1', async () => {
    const accelerationFormData: CreateAccelerationForm = {
      ...createAccelerationEmptyDataMock,
      primaryShardsCount: 1,
      replicaShardsCount: 5,
      refreshType: 'manual',
    };
    const setAccelerationFormData = jest.fn();
    const wrapper = mount(
      <IndexSettingOptions
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });

  it('renders acceleration index settings with different options2', async () => {
    const accelerationFormData: CreateAccelerationForm = {
      ...createAccelerationEmptyDataMock,
      primaryShardsCount: 5,
      replicaShardsCount: 1,
      refreshType: 'autoInterval',
      refreshIntervalOptions: { refreshWindow: 1, refreshInterval: 'second' },
      checkpointLocation: 's3://test/url',
    };
    const setAccelerationFormData = jest.fn();
    const wrapper = mount(
      <IndexSettingOptions
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });
});
