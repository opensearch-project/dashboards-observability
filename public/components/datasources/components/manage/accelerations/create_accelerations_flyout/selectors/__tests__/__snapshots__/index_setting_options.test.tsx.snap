// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index settings acceleration components renders acceleration index settings with default options 1`] = `
Array [
  <div
    className="euiText euiText--medium"
    data-test-subj="define-index-header"
  >
    <h3>
      Define Index
    </h3>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiFormRow"
    id="random_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        className="euiFormLabel euiFormRow__label"
        htmlFor="random_html_id"
      >
        Refresh type
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiPopover euiPopover--anchorDownLeft euiPopover--displayBlock euiInputPopover euiSuperSelect"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="euiPopover__anchor"
        >
          <div>
            Array [
              <input
                id="random_html_id"
                type="hidden"
                value="autoInterval"
              />,
              <div
                className="euiFormControlLayout"
              >
                <div
                  className="euiFormControlLayout__childrenWrapper"
                >
                  <span
                    className="euiScreenReaderOnly"
                    id="random_html_id"
                  >
                    Select an option: Auto (Interval), is selected
                  </span>
                  <button
                    aria-describedby="random_html_id-help-0"
                    aria-haspopup="true"
                    aria-labelledby="random_html_id random_html_id"
                    className="euiSuperSelectControl"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="button"
                  >
                    Auto (Interval)
                  </button>
                  <div
                    className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                  >
                    <span
                      className="euiFormControlLayoutCustomIcon"
                    >
                      <svg
                        aria-hidden={true}
                        className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                        focusable="false"
                        height={16}
                        role="img"
                        style={null}
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>,
            ]
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="random_html_id-help-0"
      >
        Specify how often the index should refresh, which publishes the most recent changes and make them available for search.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="random_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="random_html_id"
      >
        Refresh increments
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiFormControlLayout euiFormControlLayout--group"
      >
        <div
          className="euiFormControlLayout__childrenWrapper"
        >
          <input
            aria-describedby="random_html_id-help-0"
            aria-label="Refresh increments"
            className="euiFieldNumber euiFieldNumber--inGroup"
            id="random_html_id"
            min={1}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Refresh increments"
            type="number"
            value={15}
          />
        </div>
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <select
              className="euiSelect euiFormControlLayout__append"
              onBlur={[Function]}
              onChange={[Function]}
              onMouseUp={[Function]}
              value="minute"
            >
              <option
                key="0"
                value="minute"
              >
                minutes(s)
              </option>
              <option
                key="1"
                value="hour"
              >
                hour(s)
              </option>
              <option
                key="2"
                value="day"
              >
                day(s)
              </option>
              <option
                key="3"
                value="week"
              >
                week(s)
              </option>
            </select>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <span
                className="euiFormControlLayoutCustomIcon"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="random_html_id-help-0"
      >
        Specify how frequent the index gets updated when performing the refresh job.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="random_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="random_html_id"
      >
        Checkpoint location
      </label>
       
      <div
        className="euiText euiText--extraSmall"
      >
        <a
          className="euiLink euiLink--primary"
          href="https://github.com/opensearch-project/opensearch-spark/blob/main/docs/index.md#create-index-options"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more
          <svg
            aria-hidden={true}
            aria-label="External link"
            className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
            />
          </svg>
          <span
            className="euiScreenReaderOnly"
          >
            (opens in a new tab or window)
          </span>
        </a>
      </div>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiFormControlLayout"
      >
        <div
          className="euiFormControlLayout__childrenWrapper"
        >
          <input
            aria-describedby="random_html_id-help-0"
            aria-label="Use aria labels when no actual label is in use"
            className="euiFieldText"
            id="random_html_id"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="s3://checkpoint/location"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="random_html_id-help-0"
      >
        The HDFS compatible file system location path for incremental refresh job checkpoint.
      </div>
    </div>
  </div>,
]
`;

exports[`Index settings acceleration components renders acceleration index settings with different options1 1`] = `
Array [
  <div
    className="euiText euiText--medium"
    data-test-subj="define-index-header"
  >
    <h3>
      Define Index
    </h3>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiFormRow"
    id="random_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        className="euiFormLabel euiFormRow__label"
        htmlFor="random_html_id"
      >
        Refresh type
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiPopover euiPopover--anchorDownLeft euiPopover--displayBlock euiInputPopover euiSuperSelect"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="euiPopover__anchor"
        >
          <div>
            Array [
              <input
                id="random_html_id"
                type="hidden"
                value="autoInterval"
              />,
              <div
                className="euiFormControlLayout"
              >
                <div
                  className="euiFormControlLayout__childrenWrapper"
                >
                  <span
                    className="euiScreenReaderOnly"
                    id="random_html_id"
                  >
                    Select an option: Auto (Interval), is selected
                  </span>
                  <button
                    aria-describedby="random_html_id-help-0"
                    aria-haspopup="true"
                    aria-labelledby="random_html_id random_html_id"
                    className="euiSuperSelectControl"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="button"
                  >
                    Auto (Interval)
                  </button>
                  <div
                    className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                  >
                    <span
                      className="euiFormControlLayoutCustomIcon"
                    >
                      <svg
                        aria-hidden={true}
                        className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                        focusable="false"
                        height={16}
                        role="img"
                        style={null}
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                          fillRule="non-zero"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>,
            ]
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="random_html_id-help-0"
      >
        Specify how often the index should refresh, which publishes the most recent changes and make them available for search.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="random_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="random_html_id"
      >
        Refresh increments
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiFormControlLayout euiFormControlLayout--group"
      >
        <div
          className="euiFormControlLayout__childrenWrapper"
        >
          <input
            aria-describedby="random_html_id-help-0"
            aria-label="Refresh increments"
            className="euiFieldNumber euiFieldNumber--inGroup"
            id="random_html_id"
            min={1}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Refresh increments"
            type="number"
            value={15}
          />
        </div>
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <select
              className="euiSelect euiFormControlLayout__append"
              onBlur={[Function]}
              onChange={[Function]}
              onMouseUp={[Function]}
              value="minute"
            >
              <option
                key="0"
                value="minute"
              >
                minutes(s)
              </option>
              <option
                key="1"
                value="hour"
              >
                hour(s)
              </option>
              <option
                key="2"
                value="day"
              >
                day(s)
              </option>
              <option
                key="3"
                value="week"
              >
                week(s)
              </option>
            </select>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <span
                className="euiFormControlLayoutCustomIcon"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                    fillRule="non-zero"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="random_html_id-help-0"
      >
        Specify how frequent the index gets updated when performing the refresh job.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="random_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="random_html_id"
      >
        Checkpoint location
      </label>
       
      <div
        className="euiText euiText--extraSmall"
      >
        <a
          className="euiLink euiLink--primary"
          href="https://github.com/opensearch-project/opensearch-spark/blob/main/docs/index.md#create-index-options"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more
          <svg
            aria-label="External link"
            className="euiIcon euiIcon--small euiLink__externalIcon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 8.5a.5.5 0 1 1 1 0V12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h3.5a.5.5 0 0 1 0 1H4a1 1 0 0 0-1 1v8a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V8.5Zm-5.12.339a.5.5 0 1 1-.706-.707L13.305 2H10.5a.5.5 0 1 1 0-1H14a1 1 0 0 1 1 1v3.5a.5.5 0 1 1-1 0V2.72L7.88 8.838Z"
            />
          </svg>
          <span
            className="euiScreenReaderOnly"
          >
            (opens in a new tab or window)
          </span>
        </a>
      </div>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiFormControlLayout"
      >
        <div
          className="euiFormControlLayout__childrenWrapper"
        >
          <input
            aria-describedby="random_html_id-help-0"
            aria-label="Use aria labels when no actual label is in use"
            className="euiFieldText"
            id="random_html_id"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="s3://checkpoint/location"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="random_html_id-help-0"
      >
        The HDFS compatible file system location path for incremental refresh job checkpoint.
      </div>
    </div>
  </div>,
]
`;

exports[`Index settings acceleration components renders acceleration index settings with different options2 1`] = `
Array [
  <div
    className="euiText euiText--medium"
    data-test-subj="define-index-header"
  >
    <h3>
      Define Index
    </h3>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiFormRow"
    id="random_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        className="euiFormLabel euiFormRow__label"
        htmlFor="random_html_id"
      >
        Refresh type
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiPopover euiPopover--anchorDownLeft euiPopover--displayBlock euiInputPopover euiSuperSelect"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="euiPopover__anchor"
        >
          <div>
            Array [
              <input
                id="random_html_id"
                type="hidden"
                value="autoInterval"
              />,
              <div
                className="euiFormControlLayout"
              >
                <div
                  className="euiFormControlLayout__childrenWrapper"
                >
                  <span
                    className="euiScreenReaderOnly"
                    id="random_html_id"
                  >
                    Select an option: Auto (Interval), is selected
                  </span>
                  <button
                    aria-describedby="random_html_id-help-0"
                    aria-haspopup="true"
                    aria-labelledby="random_html_id random_html_id"
                    className="euiSuperSelectControl"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="button"
                  >
                    Auto (Interval)
                  </button>
                  <div
                    className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                  >
                    <span
                      className="euiFormControlLayoutCustomIcon"
                    >
                      <svg
                        aria-hidden={true}
                        className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                        focusable="false"
                        height={16}
                        role="img"
                        style={null}
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                          fillRule="non-zero"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>,
            ]
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="random_html_id-help-0"
      >
        Specify how often the index should refresh, which publishes the most recent changes and make them available for search.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="random_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="random_html_id"
      >
        Refresh increments
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiFormControlLayout euiFormControlLayout--group"
      >
        <div
          className="euiFormControlLayout__childrenWrapper"
        >
          <input
            aria-describedby="random_html_id-help-0"
            aria-label="Refresh increments"
            className="euiFieldNumber euiFieldNumber--inGroup"
            id="random_html_id"
            min={1}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Refresh increments"
            type="number"
            value={15}
          />
        </div>
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <select
              className="euiSelect euiFormControlLayout__append"
              onBlur={[Function]}
              onChange={[Function]}
              onMouseUp={[Function]}
              value="minute"
            >
              <option
                key="0"
                value="minute"
              >
                minutes(s)
              </option>
              <option
                key="1"
                value="hour"
              >
                hour(s)
              </option>
              <option
                key="2"
                value="day"
              >
                day(s)
              </option>
              <option
                key="3"
                value="week"
              >
                week(s)
              </option>
            </select>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <span
                className="euiFormControlLayoutCustomIcon"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                    fillRule="non-zero"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="random_html_id-help-0"
      >
        Specify how frequent the index gets updated when performing the refresh job.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="random_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="random_html_id"
      >
        Checkpoint location
      </label>
       
      <div
        className="euiText euiText--extraSmall"
      >
        <a
          className="euiLink euiLink--primary"
          href="https://github.com/opensearch-project/opensearch-spark/blob/main/docs/index.md#create-index-options"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more
          <svg
            aria-label="External link"
            className="euiIcon euiIcon--small euiLink__externalIcon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 8.5a.5.5 0 1 1 1 0V12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h3.5a.5.5 0 0 1 0 1H4a1 1 0 0 0-1 1v8a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V8.5Zm-5.12.339a.5.5 0 1 1-.706-.707L13.305 2H10.5a.5.5 0 1 1 0-1H14a1 1 0 0 1 1 1v3.5a.5.5 0 1 1-1 0V2.72L7.88 8.838Z"
            />
          </svg>
          <span
            className="euiScreenReaderOnly"
          >
            (opens in a new tab or window)
          </span>
        </a>
      </div>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        className="euiFormControlLayout"
      >
        <div
          className="euiFormControlLayout__childrenWrapper"
        >
          <input
            aria-describedby="random_html_id-help-0"
            aria-label="Use aria labels when no actual label is in use"
            className="euiFieldText"
            id="random_html_id"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="s3://checkpoint/location"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="random_html_id-help-0"
      >
        The HDFS compatible file system location path for incremental refresh job checkpoint.
      </div>
    </div>
  </div>,
]
`;
